streamlit_webrtc/models.py

```diff
@@ -94,6 +94,33 @@ def on_ended(self):
         """
 
 
+VideoFrameCallback = Callable[[av.VideoFrame], av.VideoFrame]
+QueuedVideoFramesCallback = Callable[
+    [List[av.VideoFrame]], Awaitable[List[av.VideoFrame]]
+]
+AudioFrameCallback = Callable[[av.AudioFrame], av.AudioFrame]
+QueuedAudioFramesCallback = Callable[
+    [List[av.AudioFrame]], Awaitable[List[av.AudioFrame]]
+]
+MediaEndedCallback = Callable[[], None]
+
+
+class MediaCallbackContainer:
```

Comment:
Processor classes can be sub classes of the callback container

Reply:
* A generic base class `ProcessorBase`
  * Subclass `MediaCallbackContainer`, `VideoProcessorBase`, `AudioProcessorBase`?

Reply:
Done.

The processor can also be referred as "callback container" though, here we use the unified name "processor".